apps/lockfile-explorer-web/src/parsing/lfxGraphSerializer.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { IJsonLfxDependency, IJsonLfxEntry, IJsonLfxGraph } from './JsonLfxGraph';
+import { LfxGraph, LockfileDependency, LockfileEntry } from './LfxGraph';
+
+export function serializeToJson(graph: LfxGraph): IJsonLfxGraph {
+  const jsonLfxEntries: IJsonLfxEntry[] = [];
+
+  const jsonIdByEntry: Map<LockfileEntry, number> = new Map();
+
+  function toJsonId(entry: LockfileEntry): number {
+    const result: number | undefined = jsonIdByEntry.get(entry);
+    if (result === undefined) {
+      throw new Error('Attempt to serialize disconnected object');
+    }
+    return result;
+  }
+
+  for (const entry of graph.entries) {
+    const nextIndex: number = jsonLfxEntries.length;
+
+    const jsonLfxEntry: IJsonLfxEntry = {
+      jsonId: nextIndex,
+
+      kind: entry.kind,
+      entryId: entry.entryId,
+      rawEntryId: entry.rawEntryId,
+      packageJsonFolderPath: entry.packageJsonFolderPath,
+      entryPackageName: entry.entryPackageName,
+      displayText: entry.displayText,
+      entryPackageVersion: entry.entryPackageVersion,
+      entrySuffix: entry.entrySuffix,
```

Comment:
```suggestion
      ...entry,
```

Reply:
Then destructure `entry` into those properties so you aren't repeatedly making the VM hop back to it in memory.